### ISSUE TITLE
reportCallInDefaultInitializer generates false positives for immutable classes like int, frozenset, and Decimal

### ISSUE THREAD

**Issue by etfdev** [microsoft/pyright#3757]
PoC:

```python
import decimal
from typing import FrozenSet


def f(values: FrozenSet[int] = frozenset({1, 2, 3})):
    pass


def g(value: decimal.Decimal = decimal.Decimal(0)):
    pass


def h(value: int = int(3)):
    pass
```

**To Reproduce**

With `reportCallInDefaultInitializer = "error"`:

```
$ pyright --version
pyright 1.1.227

$ pyright poc.py
/home/user/poc.py
  /home/user/poc.py:5:32 - error: Function calls and mutable objects not allowed within parameter default value expression (reportCallInDefaultInitializer)
  /home/user/poc.py:5:42 - error: Function calls and mutable objects not allowed within parameter default value expression (reportCallInDefaultInitializer)
  /home/user/poc.py:9:32 - error: Function calls and mutable objects not allowed within parameter default value expression (reportCallInDefaultInitializer)
  /home/user/poc.py:13:20 - error: Function calls and mutable objects not allowed within parameter default value expression (reportCallInDefaultInitializer)
4 errors, 0 warnings, 0 informations
Completed in 0.667sec
```

Could we teach `pyright` that constructors like `int()`, `frozenset()`, and `Decimal()` create immutable objects which are okay as default values?

Apologies if this has already been fixed in the latest version -- it's a bit hard for me to access arbitrary versions at work.

**Comment by erictraut** [comment]
The `reportCallInDefaultInitializer` check is working as designed. It is a strict-mode option, and it's somewhat opinionated, and false positives are possible. You can choose to disable the check if you think that it's too noisy for your code base. Or you can use a `# type: ignore` or `# pyright: ignore` to suppress specific instances of the error.

There's no notion of "immutable" in the Python type system, so a type checker has no way of knowing whether an object is immutable.

Given the way that default parameter arguments work in Python, I would recommend against using any complex object (anything other than `None`, a literal str/int/float value) as a default argument value. So my recommendation would be to avoid doing what you're doing in the examples above.
